### PR TITLE
add some documentation

### PR DESCRIPTION
**Pull request by kloenk**

### CODE REVIEW

File: Sources/MatrixClient/API/Request.swift [hunk 1]
```diff
@@ -7,32 +7,43 @@
 
 import Foundation
 
+/// HTTP methode to be used by the Request.
```
Comment (kiliankoe) [inline Sources/MatrixClient/API/Request.swift]:
```suggestion
/// HTTP method to be used by the request.
```

File: Sources/MatrixClient/API/Request.swift [hunk 2]
```diff
@@ -7,32 +7,43 @@
...
 public enum HttpMethod: String, CaseIterable {
     case GET
     case POST
     case PUT
     case PATCH
 
-    static var containsBody: [Self] = [.POST, .PUT, .PATCH]
+    /// Methodes which contains a body to parse.
```
Comment (kiliankoe) [inline Sources/MatrixClient/API/Request.swift]:
```suggestion
    /// Methods which contain a body to parse.
```

File: Sources/MatrixClient/API/Request.swift [hunk 3]
```diff
@@ -7,32 +7,43 @@
...
+    public static var containsBody: [Self] = [.POST, .PUT, .PATCH]
 }
 
+/// A type that can be requested on a Matrix server.
 public protocol MatrixRequest: Codable {
+    /// The type of response the Matrix server anwsers on succes.
```
Comment (kiliankoe) [inline Sources/MatrixClient/API/Request.swift]:
```suggestion
    /// The type of response the Matrix server answers with on success.
```

File: Sources/MatrixClient/API/Request.swift [hunk 4]
```diff
@@ -7,32 +7,43 @@
...
     associatedtype Response: MatrixResponse
 
+    /// Type used as extra parameter to create the request.
```
Comment (kiliankoe) [inline Sources/MatrixClient/API/Request.swift]:
```suggestion
    /// Type used as an extra parameter to create the request.
```

File: Sources/MatrixClient/API/Request.swift [hunk 5]
```diff
@@ -7,32 +7,43 @@
...
     associatedtype URLParameters
+
+    /// Function to create the URL to request, based on ``URLParameters``.
     func components(for homeserver: MatrixHomeserver, with parameters: URLParameters) throws -> URLComponents
 
+    /// The HTTP methode used by this request.
```
Comment (kiliankoe) [inline Sources/MatrixClient/API/Request.swift]:
```suggestion
    /// The HTTP method used by this request.
```

File: Sources/MatrixClient/API/Request.swift [hunk 6]
```diff
@@ -7,32 +7,43 @@
...
     static var httpMethod: HttpMethod { get }
+
+    /// True if the request requires Authentication.
```
Comment (kiliankoe) [inline Sources/MatrixClient/API/Request.swift]:
```suggestion
    /// `True` if the request requires authentication.
```

File: Sources/MatrixClient/API/Request.swift [hunk 7]
```diff
@@ -7,32 +7,43 @@
...
     static var requiresAuth: Bool { get }
+
     // TODO: rate limited property?
 }
 
 public extension MatrixRequest {
+    /// Create the URL request to get the response data for this ``MatrixRequest``.
```
Comment (kiliankoe) [inline Sources/MatrixClient/API/Request.swift]:
```suggestion
    /// Create the `URLRequest` to get the response data for this ``MatrixRequest``.
```

File: Sources/MatrixClient/API/Request.swift [hunk 8]
```diff
@@ -60,6 +71,9 @@ public extension MatrixRequest {
         return try Response(fromMatrixRequestData: data)
     }
 
+    /// Download the request data from the ``MatrixHomeserver``.
+    ///
+    /// The request do download the data is build via ``MatrixRequest/request(on:withToken:with:)``.
```
Comment (kiliankoe) [inline Sources/MatrixClient/API/Request.swift]:
```suggestion
    /// The request to download the data is built via ``MatrixRequest.request(on:withToken:with:)``.
```

File: Sources/MatrixClient/API/Request.swift [hunk 9]
```diff
@@ -79,6 +104,9 @@ public extension MatrixRequest {
         return (data, response)
     }
 
+    /// Download the request data from the ``MatrixHomeserver``.
+    ///
+    /// The request do download the data is build via ``MatrixRequest/request(on:withToken:with:)``.
```
Comment (kiliankoe) [inline Sources/MatrixClient/API/Request.swift]:
```suggestion
    /// The request to download the data is built via ``MatrixRequest.request(on:withToken:with:)``.
```

File: Sources/MatrixClient/API/Request.swift [hunk 10]
```diff
@@ -106,6 +145,7 @@ public extension MatrixRequest {
         }
     }
 
+    /// Execute the ``MatrixRequest`` returning the ``MatrixRequest/Response``.
```
Comment (kiliankoe) [inline Sources/MatrixClient/API/Request.swift:148]:
```suggestion
    /// Execute the ``MatrixRequest`` returning the ``MatrixRequest.Response``.
```

File: Sources/MatrixClient/API/Request.swift [hunk 11]
```diff
@@ -120,6 +160,7 @@ public extension MatrixRequest {
         return try parse(data: data, response: response)
     }
 
+    /// Execute the ``MatrixRequest`` returning the ``MatrixRequest/Response``.
```
Comment (kiliankoe) [inline Sources/MatrixClient/API/Request.swift:163]:
```suggestion
    /// Execute the ``MatrixRequest`` returning the ``MatrixRequest.Response``.
```

File: Sources/MatrixClient/MatrixClient.docc/Extensions/MatrixResponse.md [hunk 12]
```diff
@@ -0,0 +1,14 @@
+# ``MatrixClient/MatrixResponse``
+
+@Metadata {
+    @TechnologyRoot
+    @DocumentationExtension(mergeBehavior: append)
+}
+
+A type that that a Matrix server can response with.
```
Comment (kiliankoe) [inline Sources/MatrixClient/MatrixClient.docc/Extensions/MatrixResponse.md]:
```suggestion
A type that that a Matrix server can responsd with.
```

File: Sources/MatrixClient/MatrixClient.docc/MatrixClient.md [hunk 13]
```diff
@@ -0,0 +1,25 @@
+# ``MatrixClient``
+
+@Metadata {
+    @TechnologyRoot
+}
+
+Matrix Client with swift 5.5 async API.
```
Comment (kiliankoe) [inline Sources/MatrixClient/MatrixClient.docc/MatrixClient.md]:
```suggestion
Matrix Client with Swift 5.5 async API.
```

File: Sources/MatrixClient/MatrixClient.docc/MatrixClient.md [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+
+## Overview
+
+## Topics
+
+### Client class
```
Comment (kiliankoe) [inline Sources/MatrixClient/MatrixClient.docc/MatrixClient.md]:
```suggestion
### Client Class
```

File: Sources/MatrixClient/MatrixClient.docc/Resolve-homeserver-url.md [hunk 15]
```diff
@@ -0,0 +1,18 @@
+# Resolve Homeserver url
```
Comment (kiliankoe) [inline Sources/MatrixClient/MatrixClient.docc/Resolve-homeserver-url.md]:
```suggestion
# Resolve Homeserver URL
```

File: Sources/MatrixClient/MatrixClient.docc/Resolve-homeserver-url.md [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+
+Resolve the actuall home server url from the given well known url
```
Comment (kiliankoe) [inline Sources/MatrixClient/MatrixClient.docc/Resolve-homeserver-url.md]:
```suggestion
Resolve the actual homeserver URL from the given well-known url
```

File: Sources/MatrixClient/MatrixClient.docc/Resolve-homeserver-url.md [hunk 17]
```diff
@@ -0,0 +1,18 @@
...
+
+## Overview
+
+With ``MatrixHomeserver`` you can resolve an homeserver url usable by ``MatrixClient``.
```
Comment (kiliankoe) [inline Sources/MatrixClient/MatrixClient.docc/Resolve-homeserver-url.md]:
```suggestion
With ``MatrixHomeserver`` you can resolve a homeserver URL usable by ``MatrixClient``.
```

File: Sources/MatrixClient/MatrixClient.docc/Resolve-homeserver-url.md [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+
+## Resolving
+To resolve the homeserver url you need to init ``MatrixHomeserver`` with
+``MatrixHomeserver/init(resolve:withUrlSession:)``.
```
Comment (kiliankoe) [inline Sources/MatrixClient/MatrixClient.docc/Resolve-homeserver-url.md:11]:
```suggestion
``MatrixHomeserver.init(resolve:withUrlSession:)``.
```

File: Sources/MatrixClient/MatrixClient.docc/Resolve-homeserver-url.md [hunk 19]
```diff
@@ -0,0 +1,18 @@
...
+
+```swift
+let homesrver = try await MatrixHomeserver(resolve: "https://matrix.org/")
```
Comment (kiliankoe) [inline Sources/MatrixClient/MatrixClient.docc/Resolve-homeserver-url.md]:
```suggestion
let homeserver = try await MatrixHomeserver(resolve: "https://matrix.org/")
```

**Review by kiliankoe — COMMENTED** [review]
You wanted typo feedback 😬

File: Sources/MatrixClient/API/Request.swift (hunk 8 above)
Comment (kloenk) [reply Sources/MatrixClient/API/Request.swift]: `/` is correct here, as this is a `DocC` link

Comment (kiliankoe) [reply Sources/MatrixClient/API/Request.swift]: Ah I see!

**Pull request merged**